controller/program/tests/tests/start_rebalance.rs

```diff
@@ -0,0 +1,1530 @@
+        &lst_state_list,
+        Some(&all_pool_reserves),
+        rebalance_auth,
+        out_mint,
+        inp_mint,
+        withdraw_to,
+        actual_out_balance,
+        inp_balance,
+    );
+
+    upsert_account(
+        &mut accounts,
+        (
+            Pubkey::new_from_array(rebalance_auth),
+            Account {
+                lamports: u64::MAX,
+                owner: Pubkey::new_from_array(SYSTEM_PROGRAM_ID),
+                ..Default::default()
+            },
+        ),
+    );
+    upsert_account(
+        &mut accounts,
+        (
+            Pubkey::new_from_array(withdraw_to),
+            mock_token_acc(raw_token_acc(out_mint, withdraw_to, 0)),
+        ),
+    );
+    upsert_account(
+        &mut accounts,
+        (
+            Pubkey::new_from_array(SYSTEM_PROGRAM_ID),
+            mock_system_program_account(),
+        ),
+    );
+
+    {
+        let out_prog = Pubkey::new_from_array(builder.out_calc_prog);
+        let calc_acc = mock_prog_acc(Pubkey::new_unique());
+        upsert_account(&mut accounts, (out_prog, calc_acc));
+    }
+
+    {
+        let inp_prog = Pubkey::new_from_array(builder.inp_calc_prog);
+        let calc_acc = mock_prog_acc(Pubkey::new_unique());
+        upsert_account(&mut accounts, (inp_prog, calc_acc));
+    }
+
+    additional_accounts
+        .into_iter()
+        .for_each(|account| upsert_account(&mut accounts, account));
+
+    let instructions_account = instructions_sysvar(&instructions, 0).1;
+    upsert_account(
+        &mut accounts,
+        (
+            Pubkey::new_from_array(INSTRUCTIONS_SYSVAR_ID),
+            instructions_account,
+        ),
+    );
+
+    let rebalance_record_pda = Pubkey::new_from_array(REBALANCE_RECORD_ID);
+    upsert_account(
+        &mut accounts,
+        (rebalance_record_pda, mock_empty_rebalance_record_account()),
+    );
+
+    let initial_accounts = accounts.clone();
+    let InstructionResult {
+        program_result,
+        resulting_accounts,
+        ..
+    } = SVM.with(|svm| svm.process_instruction(&instructions[0], &accounts));
+
+    StartCaseOutcome {
+        program_result,
+        resulting_accounts,
+        initial_accounts,
+        out_mint,
+        inp_mint,
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+fn run_start_case(
+    pool: PoolState,
+    lsl: LstStateListData,
+    out_lsd: LstStateData,
+    inp_lsd: LstStateData,
+    rebalance_auth: [u8; 32],
+    out_calc_prog: [u8; 32],
+    out_calc: SvcCalcAccsAg,
+    inp_calc_prog: [u8; 32],
+    inp_calc: SvcCalcAccsAg,
+    amount: u64,
+    out_balance: u64,
+    inp_balance: u64,
+    withdraw_to: [u8; 32],
+    min_starting_out_lst: u64,
+    max_starting_inp_lst: u64,
+    include_end_rebalance: bool,
+    out_index_override: Option<u32>,
+    inp_index_override: Option<u32>,
+    out_lst_value_calc_accs_override: Option<u8>,
+    additional_accounts: impl IntoIterator<Item = PkAccountTup>,
+    error_type: Option<StartError>,
```

Comment:
we dont need a separate enum for `error_type`, refer to https://github.com/igneous-labs/inf-1.5/pull/94/files#diff-c61ac04070011cb1640e83ec2ba4d19c1c9ba446fed326137561dd3ec6eacd65